models/intermediate/field_history/int_jira__issue_calendar_spine.sql

```diff
@@ -39,39 +40,43 @@ issue_dates as (
 
     select
         issue_history_scd.issue_id,
+        issue_history_scd.source_relation,
         cast( {{ dbt.date_trunc('day', 'issue.created_at') }} as date) as created_on,
         -- resolved_at will become null if an issue is marked as un-resolved. if this sorta thing happens often, you may want to run full-refreshes of the field_history models often
-        -- if it's not resolved include everything up to today. if it is, look at the last time it was updated 
+        -- if it's not resolved include everything up to today. if it is, look at the last time it was updated
         cast({{ dbt.date_trunc('day',
             'case when issue.resolved_at is null then ' ~ dbt.current_timestamp() ~ ' else cast(issue_history_scd.valid_starting_on as ' ~ dbt.type_timestamp() ~ ') end') }}
             as date) as open_until
     from issue_history_scd
     left join {{ ref('stg_jira__issue') }} as issue
         on issue_history_scd.issue_id = issue.issue_id
+        and issue_history_scd.source_relation = issue.source_relation
 ),
 
 issue_spine as (
 
-    select 
+    select
         spine.date_day,
         issue_dates.issue_id,
+        issue_dates.source_relation,
         -- will take the table-wide min of this in the incremental block at the top of this model
         min(issue_dates.open_until) as earliest_open_until_date
 
-    from spine 
+    from spine
     join issue_dates on
         issue_dates.created_on <= spine.date_day
         and {{ dbt.dateadd('month', var('jira_issue_history_buffer', 1), 'issue_dates.open_until') }} >= spine.date_day
         -- if we cut off issues, we're going to have to do a full refresh to catch issues that have been un-resolved
-    group by 1,2
+    group by 1,2,3
 ),
 
 surrogate_key as (
 
-    select 
+    select
         date_day,
         issue_id,
-        {{ dbt_utils.generate_surrogate_key(['date_day','issue_id']) }} as issue_day_id,
+        source_relation,
+        {{ dbt_utils.generate_surrogate_key(['date_day','issue_id','source_relation']) }} as issue_day_id,
```

Comment:
I would say we should highlight in the changelog that this surrogate key is changing but looks like this model is ephemeral anyways, so I think we're good to leave it out